apiserver/facades/client/client/instanceconfig.go

```diff
@@ -70,6 +70,15 @@ func InstanceConfig(st *state.State, machineId, nonce, dataDir string) (*instanc
 	}
 	toolsList := findToolsResult.List
 
+	controllerConfig, err := st.ControllerConfig()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	caCert, ok := controllerConfig.CACert()
+	if !ok {
+		return nil, errors.New("CA certificate missing from controller config")
```

Comment:
Let's define NoCACertErr or something instead of the same string in several places.
Perhaps CACert() method itself on controllerConfig should return an error?

Reply:
Ideally it would return neither a bool nor an error, since it's not possible to have controller config without a CA certificate past bootstrap. I'm ignoring the bool now, will add a card to change the controller.Config method.